test/Integration/tpp-identity.mlir

```diff
@@ -0,0 +1,48 @@
+// RUN: tpp-opt %s -map-linalg-to-tpp -pre-bufferization -linalg-bufferize  -func-bufferize  -arith-bufferize -tensor-bufferize    -canonicalize -drop-equivalent-buffer-results -finalizing-bufferize -convert-check-to-func -convert-linalg-to-tpp -convert-tpp-to-xsmm -convert-xsmm-to-func -convert-vector-to-scf -convert-scf-to-cf |\
+// RUN: mlir-cpu-runner \
+// RUN:  -e entry -entry-point-result=void  \
+// RUN: -shared-libs=%llvmlirdir/libmlir_c_runner_utils%shlibext,%tpplibdir/libtpp_c_runner_utils%shlibext
+// 
+
+#map = affine_map<(d0, d1, d2, d3) -> (d3)>
+#map1 = affine_map<(d0, d1, d2, d3) -> (d0, d1, d2, d3)>
+
+module{
+func.func private @generate_1D_source(%width : index) -> tensor<?xf32> {
+  %init_source = tensor.empty(%width) : tensor<?xf32>
+  %source = linalg.generic {
+      indexing_maps = [affine_map<(d0) -> (d0)>],
+      iterator_types = ["parallel"]}
+      outs(%init_source : tensor<?xf32>) {
+    ^bb0(%b0 : f32):
+      %inner = linalg.index 0 : index
+      %inner_val_i32 = arith.index_cast %inner : index to i32
+      %inner_val = arith.sitofp %inner_val_i32 : i32 to f32
+      linalg.yield %inner_val :  f32
+  } -> tensor<?xf32>
+  return %source : tensor<?xf32>
+}
+
+ func.func @entry(){
+   %cst = arith.constant 64:index
+   %input_var = call @generate_1D_source(%cst): (index) -> (tensor<?xf32>)
+   %input_tensor = tensor.cast %input_var: tensor<?xf32> to tensor<64xf32>
+   %input = bufferization.to_memref %input_tensor: memref<64xf32, strided<[?], offset:? >>
```

Comment:
Ok I see why you did not use `one-shot`. You can do like this:
instead of `memref.alloc` use `tensor.empty` then prior to bufferization use: `empty-tensor-to-alloc-tensor`

Reply:
I am not using memref.alloc here, the bufferization.to_memref is the one causing problems with one-shot-bufferization and I am using it to convert tensor to memref. Can you please suggest a workaround for this?

Reply:
``` #map = affine_map<(d0) -> (d0)>
#map1 = affine_map<(d0, d1, d2, d3) -> (d3)>
#map2 = affine_map<(d0, d1, d2, d3) -> (d0, d1, d2, d3)>
module {
  func.func private @generate_1D_source(%arg0: index) -> tensor<?xf32> {
    %0 = tensor.empty(%arg0) : tensor<?xf32>
    %1 = linalg.generic {indexing_maps = [#map], iterator_types = ["parallel"]} outs(%0 : tensor<?xf32>) {
    ^bb0(%out: f32):
      %2 = linalg.index 0 : index
      %3 = arith.index_cast %2 : index to i32
      %4 = arith.sitofp %3 : i32 to f32
      linalg.yield %4 : f32
    } -> tensor<?xf32>
    return %1 : tensor<?xf32>
  }
  func.func @entry() {
    %cst = arith.constant 0.000000e+00 : f32
    %c64 = arith.constant 64 : index
    %0 = call @generate_1D_source(%c64) : (index) -> tensor<?xf32>
    %cast = tensor.cast %0 : tensor<?xf32> to tensor<64xf32>
    %1 = tensor.empty() : tensor<12x56x56x64xf32>
    %2 = linalg.generic {indexing_maps = [#map1, #map2], iterator_types = ["parallel", "parallel", "parallel", "parallel"], library_call = "tpp.identity"} ins(%cast : tensor<64xf32>) outs(%1 : tensor<12x56x56x64xf32>) {
    ^bb0(%in: f32, %out: f32):
      linalg.yield %in : f32
    } -> tensor<12x56x56x64xf32>
    %3 = tensor.empty() : tensor<12x56x56x64xf32>
    %4 = linalg.generic {indexing_maps = [#map1, #map2], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins(%cast : tensor<64xf32>) outs(%3 : tensor<12x56x56x64xf32>) {
    ^bb0(%in: f32, %out: f32):
      linalg.yield %in : f32
    } -> tensor<12x56x56x64xf32>
    check.expect_almost_eq(%2, %4, %cst) : tensor<12x56x56x64xf32>, tensor<12x56x56x64xf32>, f32
    return
  }
} ```

Reply:
`./build/bin/tpp-opt test/Integration/tpp-identity.mlir -canonicalize -empty-tensor-to-alloc-tensor -one-shot-bufferize="bufferize-function-boundaries allow-return-allocs function-boundary-type-conversion=identity-layout-map"  -canonicalize -drop-equivalent-buffer-results -finalizing-bufferize`

Reply:
this are the passes you need to bufferize.

Reply:
I have updated the test, but looks like the test fails to compile (compilation step hangs at convert-linalg-to-tpp stage).

Reply:
Then there is a bug during conversion.